scripts/ci/pre_cache.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -u
+
+# if there is no directory for this $CI_COMMIT_REF_NAME/$CI_JOB_NAME
+# create such directory and
+# copy recursively all the files from the newest dir which has $CI_JOB_NAME, if it exists
+
+# cache lives in /ci-cache/${CI_PROJECT_NAME}/${2}/${CI_COMMIT_REF_NAME}/${CI_JOB_NAME}
```

Comment:
I'm still not sure how `ci-cache` directory correlates with the cache location for our bare metal runners. We have defined `ci-cache` as the volume for Docker executor [here](https://gitlab.parity.io/parity/devops/-/blob/master/ansible/roles/gitlab-runner/templates/config.toml.j2#L27), but we don't set `cache_dir` anywhere to use that volume as the location for caching (as per [docs](https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-runners-section) for `cache_dir` setting).

Reply:
we don't use GitLab's built-in cache, we attach our own as a [rw volume ](https://gitlab.parity.io/parity/devops/-/blob/master/ansible/roles/gitlab-runner/templates/config.toml.j2#L27) to each image.

Reply:
Right, thanks, that's something that we've talked through before, but what about the first volume, `/cache`? Is it used somewhere?

Reply:
and that one is for GitLab's built-in, AFAIK it may misbehave when someone accidentally uses `cache:`.